src/extension-runners/firefox-android.js

```diff
@@ -384,6 +399,58 @@ export class FirefoxAndroidExtensionRunner {
     await adbUtils.amForceStopAPK(selectedAdbDevice, selectedFirefoxApk);
   }
 
+  async adbClearArtifactsDir() {
+    const {
+      adbUtils,
+      selectedAdbDevice,
+    } = this;
+
+    log.debug('Finding Older Artifacts');
+
+    // const check = true;
+    const check = await adbUtils.clearArtifacts(selectedAdbDevice);
+
+    if (check) {
+      log.debug('Older Artifacts Found Clearing Them');
+
+      // const del = true;
+      const del = await adbUtils.clearArtifacts(selectedAdbDevice);
+
+      if (del) {
+        log.debug('Cleared Older Artifacts');
+      } else {
+        log.debug('Coudnot Clear Older Artifacts: Fail!!!');
+      }
+
+    }
+    return;
+  }
+
+  async adbCheckArtifactsDir() {
+    const {
+      adbUtils,
+      selectedAdbDevice,
+    } = this;
+
+    log.debug('Finding Older Artifacts');
+
+    // const check = true;
+    const check = await adbUtils.checkArtifacts(selectedAdbDevice);
+
+    if (check) {
+      log.debug('Run something like web-ext --cleanup to remove those files');
+
+      throw new UsageError(
+        'Artifacts directory found please check' +
+        'Run something like web-ext --cleanup-artifacts to remove those files '
+      );
+    } else {
+      return ;
+    }
+
+
+  }
```

Comment:
Based on the logs emitted by this method, I guess that you would like to implement something like this:

- the user runs `web-ext run -t firefox-android ...` and `web-ext` detects some old artifacts dir on the target device and logs an **info** (or a **warn**) message to the user to warn the user about it (a **debug** one is not good enough because it would only be visible when web-ext is running in verbose mode, which isn't the default behavior)
- If the user wants to clear the old artifacts automatically, it does run the `web-ext` command again with the addition of the `--clean-artifacts` option (which we may actually rename to `--adb-clean-artifacts` to make it clear that it would remove the artifacts on the target device connected over adb)

I don't think that `web-ext run -t firefox-android` should fail if there are some old artifacts directories on the target device, I'll emit a warning log and leave to the user the choice to when those old artifacts should be cleared.